verifyFiles.py

```diff
@@ -0,0 +1,151 @@
+	   temp_dict.setdefault(relPath, []).append(dirDetails)
+        return temp_dict
+
+#Method populate_study_folder_dict will store the details regarding each subject folder in an ordered python dictionary. Key : Folder or file name , Value : dictionary with details of the key value
+def populate_study_folder_dict(file_path):
+	list_of_dictionaries_based_on_conditions=[]
+	#study_folders_list : List contains path to the folders contianing the study folders based on each condition and os
+	study_folders_list=read_contents_from_file(file_path)
+	for folder in study_folders_list:
+           temp_study_folder_dict=OrderedDict()
+	   #fileNamesAndDirArray is used to store the relative path of the directories and files present in the directory given as input.
+           fileNamesAndDirArray=[]
+           #file_dir_dict is an ordered  python dictionary used to store the details of individual files.Key : Relative file path, Value : Status info
+           file_dir_dict=OrderedDict()
+	   fileNamesAndDirArray=list_files_and_dirs(folder)
+	   file_dir_dict=populate_file_dir_dict(fileNamesAndDirArray,folder)
+	   temp_study_folder_dict[folder]=file_dir_dict
+	   list_of_dictionaries_based_on_conditions.append(temp_study_folder_dict)
+	return list_of_dictionaries_based_on_conditions
+
+#read_contents_from_file method is used to read the directory path containing the subject folders.
+def read_contents_from_file(fileDir): 
+# Open the file for reading.
+	with open(fileDir, 'r') as infile:
+	   data=infile.read()  # Read the contents of the file into memory.
+	   #Return a list of the lines, breaking at line boundaries.
+	   directory_list=data.splitlines()
+	return directory_list
+
+#Method generate_checksum is used for generating checksum of individual files.
+def generate_checksum(rootdir, filename):
+        hasher=hashlib.md5()
+        if os.path.isfile(os.path.join(rootdir, filename)):
+            md5_sum=file_hash(os.path.join(rootdir, filename),hasher)
+	elif os.path.isdir(os.path.join(rootdir, filename)):
+            md5_sum=directory_hash(hasher,os.path.join(rootdir, filename))
+	return md5_sum
+
+#Method file_hash is used for generating md5 checksum of a file Input: file name and hasher
+def file_hash(fileName,hasher):
+    fileContent=open(fileName)
+    while True:
+        readBuffer=fileContent.read(2**20)
+        if len(readBuffer)==0: 
+	   break
+        hasher.update(readBuffer)
+    fileContent.close()
+    return hasher.hexdigest()
+
+#Method directory_hash recursively traverses in case a directory is found, sorts the contents and creates checksum on the entire read content.
+def directory_hash(hasher, dirPath):
+    if os.path.isdir(dirPath):
+        for entry in sorted(os.listdir(dirPath)):
+            directory_hash(hasher, os.path.join(dirPath, entry))
+    elif os.path.isfile(dirPath):
+        hasher.update(file_hash(dirPath,hasher))
+    else: pass
+    return hasher.hexdigest()
```

Comment:
So if I understand well, the hash of a directory will be the hash of all the files in this directory. I don't think it's reasonable to do that. Instead, the hash of a directory should be the hash of the ordered list of entry *names* in this directory. Also, I don't think it should be recursive. Two directories will differ if and only if they contain a different list of files.